xapian-letor/tests/api_letor.cc

```diff
@@ -804,3 +804,42 @@ DEFINE_TESTCASE(err_scorer, !backend)
 
     return true;
 }
+
+//Test for TfDoclenFeature
+DEFINE_TESTCASE(tfdoclenfeature, generated)
+{
+    string db_path = get_database_path("apitest_tfdoclenfeature",
+				       db_index_one_document);
```

Comment:
This copies a pattern from the existing code, but that existing pattern isn't really right.

We're generating databases with `db_index_one_document` but in each testcase the generated database seems to be given a different name.  However these databases will all be identical, and if we use the same name when the same generator function is used (and with the same optional parameter if that's specified which it isn't here) then we can share a single generated database between all the testcases which use it.

I tend to think it's a flaw in the current harness that you have to explicitly give names to generated databases, but I haven't come up with a good way to name them automatically.

Given this is an existing issue, it probably makes sense to leave this as is, and fix the generated databases to be shareable as a separate change.

Reply:
I will try to fix that issue.